### DISCUSSION TITLE
Задание 3

### DISCUSSION THREAD

**Discussion by eprilutskaya** [ManFred1008/Amasty_Internship#5] [General]
Amasty/MyModule/Block/Form.php:37
$this->scopeConfig->isSetFlag всегда возвращает true или false, поэтому использовать ?: false не нужно

Если в методе проверяется нужно ли показывать поле qty, то и название стоит делать соответствующее, например isShowQtyInput .

Amasty/MyModule/view/frontend/templates/form.phtml:11
Если мы хотим ограничить вывод используя php, то используем два тега
```
<?php if условие :?>
что-то что должно выполниться и быть на странице
<?php endif;?>
```
```
<?php if ($block->getQty()) :?>
    <div >
        <label ...
    </div>
<?php endif;?>
```
Данные нужно получать не из глобального POST.
Благодаря наследованию (\Magento\Framework\App\Action\Action -> \Magento\Framework\App\Action\AbstractAction) в контролере доступен метод getRequest , который возвращает объект с данными запроса, оттуда мы можем получить массив данных с формы, где будет доступно sku и qty
```
$data = $this->getRequest()->getParams();
$data['sku']
```
Соблюдаем структуру if со скобочками, чтобы не сбивать код.
```
if () {
	
}
```
Вывод сообщений делаем через $this->messageManager (доступно от \Magento\Framework\App\Action\Action)
```
$this->messageManager->addErrorMessage('сообщение');
$this->messageManager->addNoticeMessage('сообщение');
$this->messageManager->addSuccessMessage('сообщение');
```
И выполняем редирект обратно на страницу, а не die.

Amasty/MyModule/Controller/Index/AddToCart.php:62
Возврат на свою страницу делаем используя resultRedirectFactory (доступно от \Magento\Framework\App\Action\AbstractAction)
`return $this->resultRedirectFactory->create()->setPath('andrewbar');`

**Comment by ManFred1008** [comment]
Большое спасибо за конструктивные замечания!
Почти всё переделал, выскочило пару ошибок - буду разбираться.

**Comment by ManFred1008** [comment]
Все замечания устранены. Сколько же всего у Магенто под капотом!
